### PR TITLE
fuzzy match the filter text

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
fuzzy matches the filter text provided by the user before limiting the results.

## How was this tested?
haven't yet. will test later tonight.

### CODE REVIEW

**Comment by Berserker66** [comment]
Am I the only one who thinks this approach is overkill? Ignoring case we should probably do, fuzzy matching fragments is going to use a lot of CPU, as you have two fuzzy components matching.

**Comment by alwaysintreble** [comment]
what about instead it compares both the provided fragment and frament.title()?

**Comment by black-sliver** [comment]
> Am I the only one who thinks this approach is overkill?

CI substring sounds like it should work for most queries, but I don't know how people will use it, so this is hard to answer. I leave it entirely up to you if you think it's worth or not. Just pointed out an obvious flaw.

**Comment by Berserker66** [comment]
My idea was to use it with mostly dungeon names or such.
`AP-1` for Factorio to make sure you got them all.
`Ice Palace` to make sure you didn't forget a chest.
That kind of thing.

**Comment by alwaysintreble** [comment]
Since these are intended to function as fragments of the actual locations even this iteration might be overkill so up to your discretion just trying to predict how users will use it and ensure desired results.

**Comment by black-sliver** [comment]
I think just searching lower() in lower() would be a lot better than the current suggestion. Lazy people, like me, might get punctuation right but skip capitalization altogether.

File: MultiServer.py [hunk 1]
```diff
@@ -1308,7 +1308,7 @@ def _cmd_missing(self, filter_text="") -> bool:
         if locations:
             names = [self.ctx.location_names[location] for location in locations]
             if filter_text:
-                names = [name for name in names if filter_text in name]
+                names = [name for name in names if filter_text.lower() in name.lower()]
```
Comment (black-sliver) [inline MultiServer.py:1311]:
```suggestion
                lower_filter_text = filter_text.lower()
                names = [name for name in names if lower_filter_text in name.lower()]
```

File: MultiServer.py [hunk 2]
```diff
@@ -1328,7 +1328,7 @@ def _cmd_checked(self, filter_text="") -> bool:
...
```
Comment (black-sliver) [inline MultiServer.py:1331]:
```suggestion
                lower_filter_text = filter_text.lower()
                names = [name for name in names if lower_filter_text in name.lower()]
```

**Review by black-sliver — COMMENTED** [review]
maybe?

**Pull request closed without merging**